app/views/layouts/application.html.erb

```diff
@@ -1,14 +1,71 @@
 <!DOCTYPE html>
 <html>
-  <head>
+<head>
     <title>HoopFinder</title>
     <%= csrf_meta_tags %>
 
     <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
-  </head>
+</head>
+<body>
+    <nav class="navbar navbar-expand-lg navbar-dark bg-primary">
+        <div class="container">
+            <%= link_to "Hoopfinder", root_path, :class => 'navbar-brand'  %>
+            <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarColor01" aria-controls="navbarColor01" aria-expanded="false" aria-label="Toggle navigation">
+                <span class="navbar-toggler-icon"></span>
+            </button>
 
-  <body>
-    <%= yield %>
-  </body>
+            <div class="collapse navbar-collapse" id="navbarColor01">
+                <ul class="navbar-nav mr-auto">
+                    <li class="nav-item">
+                        <a class="nav-link" href="#">Games <span class="sr-only">(current)</span></a>
+                    </li>
+                    <li class="nav-item">
+                        <a class="nav-link" href="#">Chat</a>
+                    </li>
+                    <li class="nav-item">
+                        <a class="nav-link" href="#">News</a>
+                    </li>
+                </ul>
+                <ul class="nav navbar-nav ml-auto">
+                    <% if user_signed_in? %>
+                        <li class="nav-item dropdown">
+                            <a class="nav-link dropdown-toggle" data-toggle="dropdown" href="#"><%= current_user.username %></a>
+                            <span class="caret"></span>
+                            <div class="dropdown-menu">
+                                <%= link_to 'Edit profile', edit_user_registration_path, :class => 'dropdown-item' %>
+                                <%= link_to "Sign Out", destroy_user_session_path, method: :delete, :class => 'dropdown-item'  %>
+                            </div>
+                        </li>
+                    <% else %>
+                        <li class="nav-item">
+                            <%= link_to "Log In", new_user_session_path, :class => 'nav-link'  %>
+                        </li>
+                        <li class="nav-item">
+                            <%= link_to "Sign Up", new_user_registration_path, :class => 'nav-link'  %>
+                        </li>
+                    <% end %>
+                </ul>
+            </div>
+        </div>
+    </nav>
+
+    <div class="container">
```

Comment:
This flash if statement will get a bit huge if you keep checking for each type of Flash.

why not check if the flash hash exists, and if it does, loop through it and display the right flash:

```
<% unless flash.empty? %>
  <% flash.each do |key, value| %>
    <%= content_tag :div, value, class: "alert alert-#{key}" %>
  <% end %>
<% end %>
```

Reply:
That's a good point, and I will make that change in a later commit. Thanks!